### PR TITLE
initial windows ESP-IDF wolfssh component setup

### PR DESCRIPTION
**Pull request by gojimmypi**
initial [setup_win.bat](https://github.com/wolfSSL/wolfssh/compare/master...gojimmypi:ESP32_component?expand=1#diff-9dda2c0b133c2b0bcb2d5b28ac8d88d97b1aac3b803ed402dba0fe73a465d9dc) for the ESP-IDF `wolfssh` and a placeholder for `setup.sh`

### CODE REVIEW

**Review by ejohnstown — APPROVED** [review]

**Comment by gojimmypi** [comment]
@ejohnstown  please stand by. I may need to fix up an additional item here before merge.

File: ide/Espressif/ESP-IDF/setup_win.bat [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+)
+
+::******************************************************************************************************
+:NO_USER_SETTINGS_RESTORE
+::******************************************************************************************************
+
+
+echo;
+echo Copying CMakeLists.txt to %WOLFSSHLIB_TRG_DIR%\
+xcopy %WOLFSSH_ESPIDFDIR%\libs\CMakeLists.txt                %WOLFSSHLIB_TRG_DIR%\                             /F
```
Comment (ejohnstown) [inline ide/Espressif/ESP-IDF/setup_win.bat:333]: wolfSSH doesn't have a CMakeLists.txt file.

File: ide/Espressif/ESP-IDF/setup_win.bat [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+:: setup some path variables
+echo;
+
+set SCRIPTDIR=%CD%
+set BASEDIR=%SCRIPTDIR%\..\..\..
+
+:: SSH
+set WOLFSSH_ESPIDFDIR=%BASEDIR%\IDE\Espressif\ESP-IDF
+
+set WOLFSSHLIB_TRG_DIR=%IDF_PATH%\components\wolfssh
```
Comment (ejohnstown) [inline ide/Espressif/ESP-IDF/setup_win.bat:79]: Can you quote the path? My path was "\Users\John Safranek", and was getting treated as multiple parameters to xcopy. (I did link my home directory to "\Users\John", which worked.)

**Review by ejohnstown — CHANGES_REQUESTED** [review]

File: ide/Espressif/ESP-IDF/setup_win.bat (hunk 1 above)
Comment (gojimmypi) [reply ide/Espressif/ESP-IDF/setup_win.bat]: the new script now includes default make files that were missing

File: ide/Espressif/ESP-IDF/setup_win.bat (hunk 2 above)
Comment (gojimmypi) [reply ide/Espressif/ESP-IDF/setup_win.bat:97]: added quotes to paths

**Comment by gojimmypi** [comment]
Hi @ejohnstown - I've removed the generation of `options.h` in favor of `#define WOLFSSL_USER_SETTINGS` and `user_settings.h`. Please check and confirm I understood and implemented as desired.

**Review by ejohnstown — APPROVED** [review]

**Comment by gojimmypi** [comment]
fyi @ejohnstown @dgarske from [ESP-IDF Getting Started](https://docs.espressif.com/projects/esp-idf/en/v4.4.1/esp32/get-started/index.html#step-5-start-a-project):

image

Still, I'm glad the extra time was spent on supporting spaces in paths.

**Pull request merged**